src/InsightContainer/index.js

```diff
@@ -0,0 +1,276 @@
+    }
+  },
+  contextGrid: {
+    padding: '0 1.25rem'
+  },
+  contextHead: {
+    fontWeight: 'bold',
+    fontSize: '1rem'
+  },
+  contextBrief: {
+    fontSize: '0.8125rem',
+    lineHeight: 2
+  },
+  linkGrid: {
+    margin: '3rem 0',
+    [breakpoints.up('md')]: {
+      margin: 0
+    }
+  }
+}));
+
+function InsightContainer({
+  loading,
+  content,
+  source,
+  title,
+  children,
+  insightActions,
+  insightContext,
+  insightLink,
+  gaEvents
+}) {
+  const classes = useStyles();
+  const {
+    handleShare,
+    handleComapre,
+    handleDownload,
+    handleShowData
+  } = insightActions;
+
+  return (
+    <Grid container spacing={4} className={classes.root}>
+      <Grid container item md={3} sm={12}>
+        <BlockLoader loading={loading}>{children[0]}</BlockLoader>
+        <TypographyLoader
+          loading={loading}
+          loader={{
+            primaryOpacity: 0.5,
+            secondaryOpacity: 1
+          }}
+          component="span"
+          className={classes.sourceGrid}
+        >
+          {source && (
+            <A className={classes.sourceLink} href={source.href}>
+              {`Source: ${source.title || source.href} `}
+            </A>
+          )}
+        </TypographyLoader>
+      </Grid>
+      <Grid container item md={5} sm={12}>
+        <Grid item>
+          <TypographyLoader
+            loading={loading}
+            loader={{
+              primaryOpacity: 0.5,
+              secondaryOpacity: 1
+            }}
+            className={classes.title}
+            variant="h5"
+          >
+            {title}
+          </TypographyLoader>
+        </Grid>
+        <Grid
+          container
+          justify="center"
+          className={classes.content}
+          style={{ width: content.width, height: content.height }}
+        >
+          <BlockLoader loading={loading}>{children[1]}</BlockLoader>
+        </Grid>
+      </Grid>
+      <Grid container item md={4} sm={12} className={classes.actionsGrid}>
+        <Grid
+          container
+          item
+          direction="row"
+          alignItems="flex-start"
+          justify="center"
+        >
+          <BlockLoader loading={loading} height="2.5rem">
+            <Actions
+              onShare={handleShare}
+              onDownload={handleDownload}
+              onShowData={handleShowData}
+              onCompare={handleComapre}
+              gaEvents={gaEvents}
+            />
+          </BlockLoader>
+        </Grid>
+        {insightContext && (
+          <Grid container item className={classes.contextGrid}>
+            <BlockLoader loading={loading}>
+              <Typography className={classes.contextHead}>
+                {insightContext.head}
+              </Typography>
+              <Typography className={classes.contextBrief}>
+                {insightContext.brief}
+              </Typography>
+            </BlockLoader>
+          </Grid>
+        )}
+        {insightLink && (
```

Comment:
Shouldn't there be at least support for two types of links? To the main country profile and to the specific section in the analysis if used in analysis?

Reply:
@kilemensi I have looked at the design. The Link section can be either to country profile or analysis profile. So we can just use props , class props to configure that